staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -486,7 +486,7 @@ type PersistentVolumeClaimSpec struct {
 	// status field of the claim.
 	// More info: https://kubernetes.io/docs/concepts/storage/persistent-volumes#resources
 	// +optional
-	Resources ResourceRequirements `json:"resources,omitempty" protobuf:"bytes,2,opt,name=resources"`
+	Resources VolumeResourceRequirements `json:"resources,omitempty" protobuf:"bytes,2,opt,name=resources"`
```

Comment:
this is a breaking change for consumers of the Go struct.
while we care more to not break the serializable API, the change should have had an action-required in the release note.

Reply:
Do we have documentation on when to use "action-required"? Would it have helped you in this case?

I keep hearing all kinds of opinions about whether Go API changes need to be mentioned at all. I added a release note, but not the "action required" because I assumed that this is reserved for changes that otherwise might silently break users on upgrades. This PR is breaking compilation in an obvious way, so developers will notice that they need to take some action.

Reply:
I think a release note is sufficient. People who happen to be using this alpha field in their go structs (who I expect will be vanishingly small) will know they need to take action when they update and their code needs a change before it will compile.

I think action required should be reserved for changes which will surprise or break people if they don't read the release notes, or which they can prepare their cluster or workloads for in advance. This is not such a change.

Reply:
FWIW, the label is documented as "Denotes a PR that introduces potentially breaking changes that require user action.".
as far as my interpretation goes, it should be added to breaking changes of any kind that require user action.

i was pinged downstream about this, not on a urgent note, and while a go build will catch the problem of the go struct that did not change for 7 years, similar would apply to a client trying to reach a REST API that changed, the user can find right away.

might be worthwhile for us to document some of these aspects in API docs.

Reply:
> it should be added to breaking changes of any kind that require user action

If we expand that to every signature change of every go function in a staging library, that section of the release notes will be so noisy it will be useless. I think it should be reserved for things which require configuration or API changes, would cause problems if they were missed, and *could* be missed in a cluster upgrade. This is not a change like that.

> while a go build will catch the problem of the go struct that did not change for 7 years

to be fair, this was a brand new alpha field, not one that had existed for years, so I sincerely doubt *anyone* was using it

Reply:
As much as I'd like to believe that the impact was low, unfortunately that's not the case here: this is a change in PersistentVolumeClaimSpec, which is a GA API. I can understand that consumers of our Go API are upset because this causes them work, but I still think it was the right thing to do because accidentally changing the Kubernetes API by sharing the same struct is worse.